<?xml version="1.0"?>

<launch>
  <master auto="start"/>

  <!-- Static transform from world to the map of the PETIT robot -->
<!--
  <node pkg="tf" type="static_transform_publisher" name="world2petit_broadcaster" args="0 0 0 0 0 0 /world /petit_map 100"/>
-->
  <!-- IMU for PETIT robot -->
<!--
  <node pkg="um6" type="um6_driver" respawn="false" name="PETIT_imu_driver" output="screen">
        <param name="port" type="str" value="/dev/ttyUSB1" />
  </node>
-->
  <!-- Fake odom from IMU -->
<!--
  <node pkg="eurobot2014_petit_nav" type="petit_odom" respawn="false" name="PETIT_fake_odom_IMU" output="screen">
  </node>
-->
  <!-- Map server for PETIT robot -->
<!--
  <node pkg="map_server" type="map_server" respawn="false" name="PETIT_map_server" output="screen" args="$(find eurobot2014_models)/map2014.yaml">
        <param name="frame_id" type="str" value="/petit_map" />
        <remap from="/map" to="/PETIT/map" />
  </node>
-->
  <!-- Laser driver => Neato XV11-->
<!--
  <node pkg="xv_11_laser_driver" type="neato_laser_publisher" respawn="false" name="PETIT_neato_laser" output="screen">
    <param name="firmware_version" value="2" />
    <param name="port" type="str" value="/dev/ttyUSB2" />
    <param name="frame_id" type="str" value="petit_neato_link" />
    
    <remap from="/scan" to="/PETIT/neato" />
  </node>
-->
  <!-- Path planner / Costmap -->
<!--  <node pkg="common_smart_nav" type="pathplanner" respawn="false" name="PETIT_pathplanner" output="screen">
    <param name="costmap_name" type="str" value="PETIT_costmap" />
    <param name="planner_name" type="str" value="PETIT_planner" />
    <param name="map_name" type="str" value="petit_map" />

    <rosparam file="$(find eurobot2014_petit_nav)/launch/nav_params.yaml" command="load" />

        <remap from="/ROBOT/goal" to="/PETIT/goal" />
        <remap from="/ROBOT/plan" to="/PETIT/plan" />
        <remap from="/ROBOT/path_done" to="/PETIT/path_done" />
        <remap from="/ROBOT/compute_path" to="/PETIT/compute_path" />
        <remap from="/ROBOT/pause" to="/PETIT/pause" />
        <remap from="/ROBOT/resume" to="/PETIT/resume" />

        <remap from="/ROBOT/get_robot_pose" to="/PETIT/get_robot_pose" />
        <remap from="/ROBOT/get_distance" to="/PETIT/get_distance" />
        <remap from="/ROBOT/get_path" to="/PETIT/get_path" />
  </node>
-->
  <!-- Path wrapper = Path follower -->
<!--  <node pkg="common_smart_nav" type="pathwrapper" respawn="false" name="PETIT_pathwrapper" output="screen">
    <param name="map_name" type="str" value="petit_map" />
    <param name="base_name" type="str" value="petit_base_link" /> 

       <remap from="/ROBOT/plan" to="/PETIT/plan" />
        <remap from="/ROBOT/get_robot_pose" to="/PETIT/get_robot_pose" />
        <remap from="/cmd_vel" to="/PETIT/cmd_vel" />
        <remap from="/resume_nav" to="/PETIT/resume_pathwrapper" />
        <remap from="/pause_nav" to="/GLOBAL/stop" />
        <remap from="path_done" to="/PETIT/path_done" />
  </node>
-->
  <!-- Low level motor driver => RoboClaw -->
<!--
  <node pkg="eurobot2014_petit_nav" type="petit_drive_roboclaw" respawn="false" name="PETIT_RoboClawDriver" output="screen">
        <remap from="/cmd_vel" to="/PETIT/cmd_vel" />
  </node>
-->
  <!-- Init pose -->
  <node pkg="eurobot2014_petit_nav" type="petit_init_pose" respawn="false" name="PETIT_initial_pose" output="screen">
        <param name="color_name" type="str" value="Yellow" />

        <remap from="/ROBOT/init_pose" to="/PETIT/init_pose" />
        <remap from="/ROBOT/cmd_vel" to="/PETIT/cmd_vel" />
        <remap from="/ROBOT/resume_pathwrapper" to="/PETIT/resume_pathwrapper" />
  </node>

  
</launch>
